aten/src/ATen/native/native_functions.yaml

```diff
@@ -1031,6 +1031,7 @@
 - func: mkldnn_linear(Tensor input, Tensor weight, Tensor? bias=None) -> Tensor
   python_module: nn
   dispatch:
+    CPU: mkldnn_linear
```

Comment:
actually after discussing it more with @bddppq - it's confusing to introduce a precedent of mixed dispatch at this level (we don't do it for cuda).

Let's move this logic to python, so that it's visible to JIT and graph transforms can observe conversion ops explicitly.

You can write the same logic of `if not x.is_mkldnn(): x.to_mkldnn()` in https://github.com/pytorch/pytorch/blob/2132ea1d8d6ca9cb2ab6c2206c7cdc9203e1908f/torch/utils/mkldnn.py#L30

and it'd be more obvious.

We'd need to expose is_mkldnn flag to python though. @bddppq can help with that

Reply:
Good idea! Though this is going to have extra conversion performance overhead (due to `to_mkldnn()` and `reshape()`), it is a better design as it is visible by JIT. I will do some benchmarking to see how big the overhead is on my side.

Reply:
Here is the PR that exposes is_mkldnn to python (and torchscript) #22386

Reply:
done in `63460ee3`, the conversion logic has been moved to `torch/utils/mkldnn.py`.